### PR TITLE
Add an option to disable opening/closing of accordion items

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-456

## Importance
Needed for the accordions used in the Entity Viewer

## Description
Added the additional prop `disabled` to the AccordionItemButton component that can be used to disable the toggling of the Accordion item.

### Other effects
- [ ] I have checked any requirements for Google Analytics
- [x] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

### CODE REVIEW

File: src/ensembl/stories/shared-components/accordion/Accordion.stories.scss [hunk 1]
```diff
@@ -19,3 +19,30 @@
...
+  color: $blue;
+}
+
+.entityViewerAccordionItem {
+  border-color: transparent;
+}
+
+.entityViewerAccordionItem + .entityViewerAccordionItem {
+  border-color: transparent;
+}
```
Comment (imransl) [inline src/ensembl/stories/shared-components/accordion/Accordion.stories.scss]: Isn't this the same as lines 35-37?

File: src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
+    ...rest
+  } = props;
 
   let styles = className;
 
   if (extendDefaultStyles) {
-    styles = classNames(defaultStyles.accordionButtonDefault, className);
+    styles = classNames(
+      defaultStyles.accordionButtonDefault,
+      { [defaultStyles.accordionButtonDefaultDisabled]: props.disabled },
```
Comment (azangru) [inline src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx]: Any reason why the name ends in DefaultDisabled, and not just Disabled?

**Comment by azangru** [comment]
The cursor changes to a pointer when moused over a disabled accordion item. It shouldn't.

File: src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx [hunk 3]
```diff
@@ -34,7 +47,7 @@ AccordionItemButton.defaultProps = {
   extendDefaultStyles: true
 };
 
-type WrapperProps = Pick<
+type WrapperProps = { disabled?: boolean } & Pick<
   DivAttributes,
   Exclude<keyof DivAttributes, keyof InjectedButtonAttributes>
 >;
```
Comment (azangru) [inline src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx]: The line below will say `const AccordionItemButtonWrapper: React.SFC<WrapperProps>`. `React.SFC` is a deprecated type (in favour of `FunctionComponent`), and the type annotation of `AccordionItemButtonWrapper` is likely unnecessary.

Comment (azangru) [reply src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx]: P.S.: there are several examples of React.SFC in our codebase — all in accordion component. Could you please weed them out?

File: src/ensembl/stories/shared-components/accordion/Accordion.stories.scss (hunk 1 above)
Comment (msmanoj) [reply src/ensembl/stories/shared-components/accordion/Accordion.stories.scss]: Yes, but this is needed to override the default styles.

File: src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx (hunk 2 above)
Comment (msmanoj) [reply src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx]: I was thinking that the disabled button has a default style which can be overridden. I have renamed it to be just Disabled as it sounds better.

File: src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx (hunk 3 above)
Comment (msmanoj) [reply src/ensembl/src/shared/components/accordion/components/AccordionItemButton.tsx]: That's done

File: src/ensembl/stories/shared-components/accordion/Accordion.stories.scss (hunk 1 above)
Comment (azangru) [reply src/ensembl/stories/shared-components/accordion/Accordion.stories.scss]: But is the rule on 35-37 actually doing anything?

Comment (msmanoj) [reply src/ensembl/stories/shared-components/accordion/Accordion.stories.scss]: No, it doesn't. I've removed it.

**Pull request merged**